pkg/apis/batch/types.go

```diff
@@ -160,6 +173,10 @@ type JobStatus struct {
 	// The number of pods which reached phase Failed.
 	// +optional
 	Failed int32
+
+	// The number of pods which reached phase Failed and deleted.
```

Comment:
Not needed, the above `Failed` field should hold information about all the failed pods. Currently it's the job controller that is broken, see https://github.com/kubernetes/kubernetes/issues/28486 for details.